### PR TITLE
Script/Spell: fix Midsummer's Juggling Torch

### PR DESCRIPTION
**Pull request by Meramaf**
## Changes Proposed:
-  This Fix adressed to Achievement "Juggle 40 torches in 15 seconds in Dalaran." 
https://www.wowhead.com/achievement=272/torch-juggler

## How to Test the Changes:

1. .additem 34599 40
2. .q a 11657 .q c 11657 .q rew 11657
3. .tele dalaran
4. cast item on self

### CODE REVIEW

**Comment by Meramaf** [comment]
Tested and work

**Comment by Maelthyrr** [comment]
Tested and work fine.

<details>

WoWScrnShot_062822_080925

</details>

File: src/server/scripts/Events/midsummer.cpp [hunk 1]
```diff
@@ -507,6 +508,32 @@ class spell_midsummer_juggling_torch : public SpellScript
...
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo({ SPELL_GIVE_TORCH });
+    }
+
+    void HandleDummy(SpellEffIndex /*effIndex*/)
+    {
+        Player* player = GetHitPlayer();
+        if (!player)
+            return;
```
Comment (IntelligentQuantum) [inline src/server/scripts/Events/midsummer.cpp:525]: Bracket

File: src/server/scripts/Events/midsummer.cpp [hunk 2]
```diff
@@ -507,6 +508,32 @@ class spell_midsummer_juggling_torch : public SpellScript
...
+
+        if (player->GetQuestStatus(QUEST_TORCH_CATCHING_A) == QUEST_STATUS_REWARDED || player->GetQuestStatus(QUEST_TORCH_CATCHING_H) == QUEST_STATUS_REWARDED)
+            player->CastSpell(player, SPELL_GIVE_TORCH);
```
Comment (IntelligentQuantum) [inline src/server/scripts/Events/midsummer.cpp:528]: bracket

**Review by acidmanifesto — CHANGES_REQUESTED** [review]
please add the request missing brackets from the reviewer.

File: data/sql/updates/pending_db_world/juggling_torch_fix.sql [hunk 3]
```diff
@@ -0,0 +1,3 @@
+DELETE FROM `spell_script_names` WHERE `spell_id` = '45644';
```
Comment (Nefertumm) [inline data/sql/updates/pending_db_world/juggling_torch_fix.sql:1]:
```suggestion
DELETE FROM `spell_script_names` WHERE `spell_id` = 45644;
```
Please don't use ' around numbers unless the column data is string

File: src/server/scripts/Events/midsummer.cpp [hunk 4]
```diff
@@ -453,7 +453,8 @@ enum eJuggle
 
     SPELL_TORCH_CHECK           = 45644,
     SPELL_GIVE_TORCH            = 45280,
-    QUEST_CHECK                 = 11937,
+    QUEST_TORCH_CATCHING_A           = 11657,
+    QUEST_TORCH_CATCHING_H           = 11923
```
Comment (Nefertumm) [inline src/server/scripts/Events/midsummer.cpp:457]: Please, align correctly these lines :)

**Review by acidmanifesto — APPROVED** [review]

**Comment by acidmanifesto** [comment]
Thank you for the PR

**Pull request merged**